packages/backend-core/src/context/mainContext.ts

```diff
@@ -394,3 +394,20 @@ export function setFeatureFlags(key: string, value: Record<string, any>) {
   context.featureFlagCache ??= {}
   context.featureFlagCache[key] = value
 }
+
+export function getTableForView(viewId: string): Table | undefined {
+  const context = getCurrentContext()
+  if (!context) {
+    return
+  }
+  return context.viewToTableCache?.[viewId]
+}
+
+export function setTableForView(viewId: string, table: Table) {
+  const context = getCurrentContext()
+  if (!context) {
+    return
+  }
+  context.viewToTableCache ??= {}
+  context.viewToTableCache[viewId] = table
+}
```

Comment:
Because we plan to use the `ViewV2` to get to the `Table` lot of times per request, I've decided to cache that lookup in the context so it's free every time after the first. This is nicer than passing both a `Table` and a `ViewV2` around everywhere.